testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeSelectCompatibility.java

```diff
@@ -0,0 +1,78 @@
+ * limitations under the License.
+ */
+package io.trino.tests.product.deltalake;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.tempto.assertions.QueryAssert.Row;
+import io.trino.testng.services.Flaky;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.tests.product.TestGroups.DELTA_LAKE_DATABRICKS;
+import static io.trino.tests.product.TestGroups.DELTA_LAKE_OSS;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.deltalake.util.DeltaLakeTestUtils.DATABRICKS_COMMUNICATION_FAILURE_ISSUE;
+import static io.trino.tests.product.deltalake.util.DeltaLakeTestUtils.DATABRICKS_COMMUNICATION_FAILURE_MATCH;
+import static io.trino.tests.product.utils.QueryExecutors.onDelta;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static org.testng.Assert.assertEquals;
+
+public class TestDeltaLakeSelectCompatibility
+        extends BaseTestDeltaLakeS3Storage
+{
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
+    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    public void testPartitionedSelectSpecialCharacters()
+    {
+        String tableName = "test_dl_partitioned_select_special" + randomNameSuffix();
+
+        onDelta().executeQuery("" +
+                "CREATE TABLE default." + tableName +
+                "         (a_number INT, a_string STRING)" +
+                "         USING delta " +
+                "         PARTITIONED BY (a_string)" +
+                "         LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "'");
+
+        try {
+            onDelta().executeQuery("INSERT INTO default." + tableName + " VALUES (1,'al=a'), (2, 'kot+a'), (3, 'osl a')");
+            onTrino().executeQuery("INSERT INTO delta.default." + tableName + " VALUES (3, 'ps=a'), (4, 'bobr %a'), (4, 'lw a'), (5, 'jez +a')");
+
+            List<Row> expectedRows = ImmutableList.of(
+                    row(1, "al=a"),
+                    row(2, "kot+a"),
+                    row(3, "osl a"),
+                    row(3, "ps=a"),
+                    row(4, "bobr %a"),
+                    row(4, "lw a"),
+                    row(5, "jez +a"));
+
+            assertThat(onDelta().executeQuery("SELECT * FROM default." + tableName))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + tableName))
+                    .containsOnly(expectedRows);
+            String deltaFilePath = (String) onDelta().executeQuery("SELECT input_file_name() FROM default." + tableName + " WHERE a_number = 1").getOnlyValue();
```

Comment:
> it looks like input_file_name returns the URI encoded path (created https://github.com/delta-io/delta/issues/1517 for this), so the test will likely continue to fail. 

Please add the code comment with the issue link.

> We can probably skip comparing with the filename obtained from Spark here, and only check that Trino paths are internally compatible?

You can add two other assertions. The 1st is `Assertions.assertThat(deltaFilePath).isNotEqualTo(trinoFilePath);` and the 2nd is verifying URI decoded path of `deltaFilePath` is equals to `trinoFilePath`.